tests/helpers.py

```diff
@@ -7,7 +7,18 @@
 def step_env(env, n=10, render=True):
     env.reset()
     for _ in range(n):
-        _, _, done, _ = env.step(env.action_space.sample())
+        a = env.action_space.sample()
+        a_copy = a if isinstance(a, int) else a.copy()
+        _, _, done, _ = env.step(a)
+
+        # Check for side effects
+        if isinstance(a, (int, float)):
+            assert a == a_copy,\
+            "Action was modified by environment!"
+        else:
+            assert a.all() == a_copy.all(),\
```

Comment:
Don't assert inside test helper functions unless the helper function name makes it crystal clear that it's a *test* and not a *function*. This helper is a function.

It makes it difficult for the test author to predict why the test might fail. It also bundles together a bunch of test cases into one function, rather than making them individual test cases.